draft-vkrasnov-h2-compression-dictionaries.md

```diff
@@ -138,58 +182,87 @@ The SET_COMPRESSION_CONTEXT frame (type=0xTBA).
-and passed as is.
-A special Context ID of 254 indicates the stream SHALL NOT be compressed with
-any dictionary, and it MUST NOT be used as dictionary for other streams.
+: an 8-bit context ID that indicates the compression context for the stream.
+If the frame is ommited, then the context value is assumed to be 0.
+The allowed context values are 0 through 255.
+A special context ID of 255 indicates the stream SHALL NOT be cross-compressed.
 
 ### The SET_DICTIONARY Frame
 
-The SET_DICTIONARY frame (type=0xTBA).
+The SET_DICTIONARY frame (type=0xTBA) contains one to many Dictionary-Entry.
 
 ~~~~
-+-------------+-------------+
-| Dict ID (8) |   Size (8)  |
-+-------------+-------------+
++---------------+---------------+
+|   Dictionary-Entry (+)    ...
++---------------+---------------+
 ~~~~
 
-The SET_DICTIONARY frame can be sent from the server to the client, on any client
-initiated stream in the open or half-closed (remote) states. The SET_DICTIONARY
-frame MUST precede any DATA frames on that stream. The SET_DICTIONARY frame SHOULD
-be followed by sufficient DATA frames to fill Size octets after decompression,
-even if an RST frame was received for that stream. If not enough DATA was sent,
-the Dictionary for the given ID is considered uninitialized.
+A Dictionary-Entry field is encoded as follows:
+
+~~~~
++-------------------------------+
+|       Dictionary-ID (8)       |
++---+---------------------------+
+| P |        Size (7+)          |
++---+---------------------------+
+| E?|      Truncate? (7+)       |
++---+---------------------------+
+|           Offset? (8+)        |
++-------------------------------+
 
-More than one SET_DICTIONARY frames MAY be sent on a given stream.
+~~~~
 
-The SET_DICTIONARY frame contains the following fields:
+The SET_DICTIONARY frame can be sent from the server to the client, on any
+client initiated stream in the open or half-closed (remote) states. The
+SET_DICTIONARY frame MUST precede any DATA frames on that stream. The
+SET_DICTIONARY frame SHOULD be followed by sufficient DATA frames to build the
+dictionaries.
+If a RST frame was received for the stream before sufficient DATA was sent, the
+dictionaries are reset.
 
-Dict ID:
+The Dictionary-Entry contains the following fields:
+
+Dictionary-ID:
 : an 8-bit ID, indicates the dictionary. MUST be lower than the value agreed by
-the SETTINGS_COMPRESSION_SETTINGS setting.
+the SETTINGS_COMPRESSION setting.
 
 Size:
-: Indicates how many octets (as a power of 2) of the given stream will be used.
-If Size is greater than the length of the transmitted data, then all of the data
-will be used.
-
-The SET_DICTIONARY frame defines the following flag:
+: Indicates how many octets of the stream will be used for the dictionary.
+Size is represented as an integer with 7-bit prefix (see {{RFC1951}},
+Section 5.1). If P is set, the actual number of octets to use is 2 to the power
```

Comment:
I don't think that's the right RFC reference

Reply:
no its not :)